variables.tf

```diff
@@ -75,6 +75,12 @@ variable "runtime" {
   description = "The runtime in which the function will be executed."
 }
 
+variable "secret_environment_variables" {
+  type        = list(map(string))
+  default     = []
+  description = "A list of maps which contains key, project_id (only project number is supported by this field), secret_id (not the full secret name) and version to assign to the function as a set of secret environment variables."
```

Comment:
I think the provider docs maybe out of data regarding the project num requirement. Could you test with project_id?
Per https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#secretenvvar
`Project identifier (preferrably project number but can also be the project ID) of the project that contains the secret.`

Reply:
secret_id is also a bit confusing due to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/secret_manager_secret#id being `projects/{{project}}/secrets/{{secret_id}}`

Lets show an example in the description.

Reply:
Yes, you're right. I've just tested the configuration of cloud_function and secrets environment variables by using the non-numeric value of project_id and it works!
I'll change the description for this variable.

As for the value of `secret_id`: can I change it to, for example, `secret_name`?

Reply:
Yeah name makes sense and let's provide an example of the expected map too.